src/crystal/system/win32/thread_local.cr

```diff
@@ -0,0 +1,30 @@
+require "c/fibersapi"
+
+class Thread
+  struct Local(T)
+    @key : LibC::DWORD
+
+    def initialize
+      {% raise "T must be a Reference or Pointer" unless T < Reference || T < Pointer %}
+      @key = LibC.FlsAlloc
+      raise RuntimeError.from_winerror("FlsAlloc: out of indexes") if @key == LibC::FLS_OUT_OF_INDEXES
+    end
+
+    def initialize(&destructor : Proc(T, Nil))
+      {% raise "T must be a Reference or Pointer" unless T < Reference || T < Pointer %}
+      @key = LibC.FlsAlloc(destructor.unsafe_as(Proc(Void*, Nil)))
```

Comment:
```suggestion
      @key = LibC.FlsAlloc(destructor.unsafe_as(LibC::FLS_CALLBACK_FUNCTION))
```

Reply:
Nope:

```
 15 | @key = LibC.FlsAlloc(destructor.pointer.as(LibC::FLS_CALLBACK_FUNCTION))
                           ^
Error: can't cast Pointer(Void) to Proc(Pointer(Void), Nil)
```

Reply:
Ok I guess we'll have to use `unsafe_as` then.
My main point is to reuse the type declaration `LibC::FLS_CALLBACK_FUNCTION` from the lib binding instead of duplicating it. I see you have already applied that 👍